internal/controller/csiaddons/persistentvolumeclaim_controller.go

```diff
@@ -322,6 +324,39 @@ func (r *PersistentVolumeClaimReconciler) determineScheduleAndRequeue(
 	return "", ErrScheduleNotFound
 }
 
+func (r *PersistentVolumeClaimReconciler) storageClassEventHandler() handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(
+		func(ctx context.Context, obj client.Object) []reconcile.Request {
+			ok := false
+			obj, ok = obj.(*storagev1.StorageClass)
+			if !ok {
+				return nil
+			}
+
+			// get all PVCs with the same storageclass.
+			pvcList := &corev1.PersistentVolumeClaimList{}
+			err := r.List(ctx, pvcList, client.MatchingFields{"spec.storageClassName": obj.GetName()})
+			if err != nil {
+				log.FromContext(ctx).Error(err, "Failed to list PVCs")
+
+				return nil
+			}
+
+			var requests []reconcile.Request
+			for _, pvc := range pvcList.Items {
+				requests = append(requests, reconcile.Request{
```

Comment:
is there a need to add the PVC to the requests if it already have an annotation? reducing the number of requests that need reconciling probably reduces the load the controller causes

Reply:
it looks like we will skip these pvcs, won't this cause a problem when an annotation is deleted on the storageclass and the controller-added annotation exists on the PVC, IMO we need an owner to identify who added the annotation is it by the controller or not.

Reply:
Yes, agreed. An annotation with the owner of the schedule would be useful for automatic updating and so on.